WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -541,7 +541,7 @@ class AbstractPostListViewController: UIViewController,
         let posts = try result.map { try coreDataStack.mainContext.existingObject(with: $0) }
         let hasMore = result.count >= number
 
-        return (posts, hasMore)
+        return (posts.deduplicated(by: \.postHash), hasMore)
```

Comment:
This leaves a period of time where database has two posts with the same `postID` – invalid state. I suggest fixing it on the `PostService/getPostWithID` level where it decides when to create a new or update the existing post.

```
AbstractPost *post = [blog lookupPostWithID:postID inContext:self.managedObjectContext];
```

I suggest updating the lookup with a `foreignID` (or whatever you want to call it).

I tested passing a foreign ID using the `metadata` field (only tested wp.com).

Request:

```
{
  "metadata": [
    {
      "key": "wp-jp-app-foreign-id",
      "operation": "update",
      "value": "x-coredata://ED210C7A-B958-4371-8193-2B708EB122B5/Post/p1218"
    }
  ],
  "format": "standard",
  "content": "<!-- wp:paragraph -->\n<p>1</p>\n<!-- /wp:paragraph -->",
  "status": "draft",
  "title": "Off-04",
  "type": "post",
  "author": 235722583
}
```

Response:


```json
{
  "ID": 713,
  "site_ID": 231378612,
  // ...
  "metadata": [
    {
      "id": "1782",
      "key": "wp-jp-app-foreign-id",
      "value": "x-coredata://ED210C7A-B958-4371-8193-2B708EB122B5/Post/p1218"
    }
  ],
  // ...
}
```

The list request `/rest/v1.2/sites/*/posts` also return the posts _with_ `wp-jp-app-foreign-id` key.

**Note**: I used `objectID` as an example, but it is probably the best type of ID we could send. It would probably be best to generate a `UUID`, but it'll require database schema changes.